test/support/struct_helper.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+module StructHelper
+  def create_singleton_struct(name, *fields)
```

Comment:
That's because of https://github.com/zendesk/samson/blob/master/app/models/changeset.rb#L83. We need the result of `GITHUB.compare` to be serializable for `Rails.cache`. `Struct.new` without a class name is not serializable. Also referencing https://github.com/zendesk/samson/commit/5b61dc27228e652e15ee9774382ec2fbc02a0bf7. 

Hmm.. Since `GITHUB.compare` returns a `Sawyer::Resource`, I can probably do something like:

```
agent = Sawyer::Agent.new('')
message = Sawyer::Resource.new(agent, message: 'some message')
comparison = Sawyer::Resource.new(agent, commits: [message])
```

Would prefer to do it as a separate refactor PR together with `changeset_test.rb` and `pull_request_test.rb`. Let me know if you want me to fix it in this PR.

The other 2 issues are corrected. Thanks!